bug: import watch history from youtube doesnt work

### Steps to reproduce

1. Open the settings
2. Try to import watch history from youtube
3. See the error

### Expected behavior

Imports the watch history

### Actual behavior

Gives error

### LibreTube version

latest version

### Android version

Android 7

### Other details

Log:

kotlinx.serialization.MissingFieldException: Field 'titleUrl' is required for type with serial name 'com.github.libretube.obj.YouTubeWatchHistoryFileItem', but it was missing at path: $[4413]
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(SourceFile:147)
	at okio.Options$Companion.decodeSerializableElement(SourceFile:11)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableElement(SourceFile:46)
	at kotlinx.serialization.internal.CollectionLikeSerializer.readElement(SourceFile:8)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.merge(SourceFile:30)
	at kotlinx.serialization.internal.AbstractCollectionSerializer.deserialize(SourceFile:1)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(SourceFile:92)
	at kotlin.io.TextStreamsKt.decodeByReader(SourceFile:27)
	at org.jsoup.Jsoup.decodeFromStream(SourceFile:31)
	at com.github.libretube.helpers.BackupHelper.importWatchHistory(SourceFile:125)
	at com.github.libretube.ui.preferences.BackupRestoreSettings$getWatchHistoryFile$1$1.invokeSuspend(SourceFile:48)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(SourceFile:9)
	at kotlinx.coroutines.DispatchedTask.run(SourceFile:113)
	at androidx.work.impl.WorkerWrapper$1.run(SourceFile:77)
	at kotlinx.coroutines.scheduling.TaskImpl.run(SourceFile:3)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(SourceFile:96)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@fde4d74, Dispatchers.IO]
Caused by: kotlinx.serialization.MissingFieldException: Field 'titleUrl' is required for type with serial name 'com.github.libretube.obj.YouTubeWatchHistoryFileItem', but it was missing
	at kotlinx.serialization.internal.PlatformKt.throwMissingFieldException(SourceFile:115)
	at com.github.libretube.obj.YouTubeWatchHistoryFileItem.<init>(SourceFile:1)
	at com.github.libretube.obj.YouTubeWatchHistoryFileItem$$serializer.deserialize(SourceFile:1)
	at com.github.libretube.obj.YouTubeWatchHistoryFileItem$$serializer.deserialize(SourceFile:2)
	at kotlinx.serialization.json.internal.StreamingJsonDecoder.decodeSerializableValue$1(SourceFile:92)
	... 15 more
    

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

--- a/app/src/main/java/com/github/libretube/helpers/ImportHelper.kt
+++ b/app/src/main/java/com/github/libretube/helpers/ImportHelper.kt
@@ -162,16 +162,17 @@ object ImportHelper {
             }
 
             ImportFormat.FREETUBE -> {
-                val playlistFile = activity.contentResolver.openInputStream(uri)?.use { inputStream ->
-                    val text = inputStream.bufferedReader().readText()
-                    runCatching {
-                        text.lines().map { line ->
-                            JsonHelper.json.decodeFromString<FreeTubeImportPlaylist>(line)
-                        }
-                    }.getOrNull() ?: runCatching {
-                        listOf(JsonHelper.json.decodeFromString<FreeTubeImportPlaylist>(text))
-                    }.getOrNull()
-                }
+                val playlistFile =
+                    activity.contentResolver.openInputStream(uri)?.use { inputStream ->
+                        val text = inputStream.bufferedReader().readText()
+                        runCatching {
+                            text.lines().map { line ->
+                                JsonHelper.json.decodeFromString<FreeTubeImportPlaylist>(line)
+                            }
+                        }.getOrNull() ?: runCatching {
+                            listOf(JsonHelper.json.decodeFromString<FreeTubeImportPlaylist>(text))
+                        }.getOrNull()
+                    }
 
                 val playlists = playlistFile.orEmpty().map { playlist ->
                     // convert FreeTube videos to list of string
@@ -287,7 +288,7 @@ object ImportHelper {
                     JsonHelper.json.decodeFromStream<List<YouTubeWatchHistoryFileItem>>(it)
                 }
                     .orEmpty()
-                    .filter { it.activityControls.contains("YouTube watch history") && it.subtitles.isNotEmpty() }
+                    .filter { it.activityControls.contains("YouTube watch history") && it.subtitles.isNotEmpty() && it.titleUrl.isNotEmpty() }
                     .reversed()
                     .map {
                         val videoId = it.titleUrl.substring(it.titleUrl.length - 11)

--- a/app/src/main/java/com/github/libretube/obj/YouTubeWatchHistoryFileItem.kt
+++ b/app/src/main/java/com/github/libretube/obj/YouTubeWatchHistoryFileItem.kt
@@ -4,10 +4,10 @@ import kotlinx.serialization.Serializable
 
 @Serializable
 data class YouTubeWatchHistoryFileItem(
-    val header: String,
-    val time: String,
-    val title: String,
-    val titleUrl: String,
+    val header: String = "",
+    val time: String = "",
+    val title: String = "",
+    val titleUrl: String = "",
     val activityControls: List<String> = emptyList(),
     val products: List<String> = emptyList(),
     val subtitles: List<YouTubeWatchHistoryChannelInfo> = emptyList()

